### PR TITLE
Update all Ruby linters to work consistently with bundler

### PR DESCRIPTION
**Pull request by rrosenblum**
This is in response to #1403

### CODE REVIEW

**Comment by rrosenblum** [comment]
@w0rp to clarify, are you wanting me to implement `ale#handlers#haskell_stack#EscapeExecutable` for ruby?

```vim
function! ale#handlers#haskell_stack#EscapeExecutable(executable, stack_exec) abort
    let l:exec_args = a:executable =~? '^stack$'
    \   ? ' exec ' . ale#Escape(a:stack_exec) . ' --'
    \   : ''
    return ale#Escape(a:executable) . l:exec_args
endfunction
```

**Comment by rrosenblum** [comment]
@w0rp I am having some trouble implementing the `EscapeExecutable` function the same way it is written in #1851.

The problem is that by using `EscapeExecutable` inside of `GetExecutable` causes `ale#engine#IsExecutable` to not think the command is executable. This appears to be caused by the command now being made up of multiple words `bundle exec rubocop` rather than a single command `bundle`. I have tried using different escaping, but I haven't found anything that causes `IsExecutable` to pass. What is the best way to move forward with this change?

File: ale_linters/ruby/brakeman.vim [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
+call ale#Set('ruby_brakeman_options', '')
+call ale#Set('ruby_brakeman_executable', 'brakeman')
+
 let g:ale_ruby_brakeman_options =
 \   get(g:, 'ale_ruby_brakeman_options', '')
 
+function! ale_linters#ruby#brakeman#GetExecutable(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'ruby_brakeman_executable')
+    return ale#handlers#ruby#EscapeExecutable(l:executable, 'brakeman')
```
Comment (w0rp) [inline ale_linters/ruby/brakeman.vim]: The answer here is that you should just return the executable here wihout escaping it, and you should use `ale#handlers#ruby#EscapeExecutable` in `GetCommand` below. The reason being that `executable_callback` should return the path to the program that will be run, and `command_callback` should return a command to run, where the part of the command with an executable name has to be escaped.

File: ale_linters/ruby/brakeman.vim [hunk 2]
```diff
@@ -27,20 +35,22 @@ function! ale_linters#ruby#brakeman#Handle(buffer, lines) abort
...
+    return ale#Escape(l:executable)
+    \    . ' -f json -q '
     \    . ale#Var(a:buffer, 'ruby_brakeman_options')
     \    . ' -p ' . ale#Escape(l:rails_root)
 endfunction
 
 call ale#linter#Define('ruby', {
 \    'name': 'brakeman',
-\    'executable': 'brakeman',
+\    'executable_callback': 'ale_linters#ruby#brakeman#GetExecutable',
```
Comment (w0rp) [inline ale_linters/ruby/brakeman.vim]: When the callbacks for `executable_callback` just `return ale#Var(a:buffer, variable_name)`, you can write `executable_calback: ale#VarFunc(variable_name)` instead to get the same result.

File: ale_linters/ruby/brakeman.vim (hunk 1 above)
Comment (rrosenblum) [reply ale_linters/ruby/brakeman.vim]: Ok, cool. That is what I was thinking, but I wanted to check with you first.

**Comment by rrosenblum** [comment]
@w0rp this is ready for review again.

**Comment by w0rp** [comment]
The callbacks only need to be adjusted slightly. I'll sort that out and then merge this.

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by stefansedich** [comment]
This is great, but one thing I want to be able to do is run `bundle` when there is a project Gemfile and fall back to my locally installed rubocop if I am editing a random ruby file without a Gemfile. Is there a nice way that I could achieve this currently?

**Comment by rrosenblum** [comment]
Executable fallback would be useful. I don't think that there is a way to get it working with the current implementation. There is some code that checks to make sure the executable command is executable so it might be possible to add this functionality. What are your thoughts @w0rp?

**Pull request merged**